testnet/stacks-node/src/run_loop/neon.rs

```diff
@@ -356,15 +359,19 @@ impl RunLoop {
             // wait for the p2p state-machine to do at least one pass
             debug!("Wait until we reach steady-state before processing more burnchain blocks...");
             // wait until it's okay to process the next sortitions
-            let ibd =
-                pox_watchdog.pox_sync_wait(&burnchain_config, &burnchain_tip, burnchain_height);
+            let ibd = pox_watchdog.pox_sync_wait(
```

Comment:
With the changes you're trying to make, it may make more sense to refactor the body of the top-level loop here into its own method.  Then, it can return `burnchain_error::CoordinatorClosed` at any preemption point in response to `should_keep_running` getting set to `false` by the `ctrlc` thread, and the outermost caller can take care of shutting down all of the relevant threads.

Reply:
Addressed with latest